### ISSUE TITLE
The goal `verify` does not fail, although the file does not end with a new line.

### ISSUE THREAD

**Issue by vrastov** [Ekryd/sortpom#445]
See pull rquest in my fork https://github.com/vrastov/sortpom/pull/1 

Maybe I misunderstood the description `<endWithNewline>`.

**Comment by Ekryd** [comment]
Hi! I'll have a look 👍

**Comment by Ekryd** [comment]
The verify goal does only look at the xml elements themselves:
_**verify** only sorts the current pom.xml file if the xml elements are unsorted. This goal ignores text formatting (such as indentation and line breaks) when it verifies if the pom is sorted or not._

So endWithNewLine should not affect the verify goal at all (for the verification part).

**Comment by vrastov** [comment]
I used the feature "verify  check formatting " :)

Okay but why:

https://github.com/vrastov/sortpom/pull/3

**Comment by Ekryd** [comment]
I'm a bit confused. Let me try to explain and see where it takes us :-) . 
There are two goals: 'sort' and 'verify'

`sort` will sort the pom file. The sort goal has a bunch of configurations regarding the sorting (e.g. `endWithNewLine` or `keepBlankLines` etc.). This goal is the one that you usually use as a developer.

`verify` will **only** check if the xml elements are unsorted. If the elements are unsorted, then the `verifyFail` configuration will determine what will happen:
If unsorted and `verifyFail` is 'Stop' then the build would terminate. This can be used in a CI pipeline to enforce each developer to sort before they checkin.
If unsorted and `verifyFail` is 'Warn' then the build would continue but warnings would be generated. Again, this is mostly useful in a CI pipeline
If unsorted and `verifyFail` is 'Sort' then the pom is sorted exactly as the sort goal would do. All the parameters that are applicable for 'sort' would be used during the actual sorting. This can be used if you want a _lesser_ control of the sorting, i.e. just resort everything whenever an xml element is placed in the wrong order.

To sum it up, the `verify` goal is useful for building really strict Continuous Integration pipelines or when you want to keep a personally formatted pom file, but ensure that xml elements are in the right order.

**Comment by vrastov** [comment]
I need a strict continuous integration pipeline.

I like the behavior of https://github.com/vrastov/sortpom/pull/3 and https://github.com/vrastov/sortpom/pull/4 and the rest.

And I would like to control the presence of a blank line at the end of the file, https://medium.com/@alexey.inkin/how-to-force-newline-at-end-of-files-and-why-you-should-do-it-fdf76d1d090e

**Comment by vrastov** [comment]
I think I understand what you mean, my examples are slightly different.

Added to https://github.com/vrastov/sortpom/pull/1

```xml
  <build>
    <plugins>

      <plugin>
        <groupId>com.github.ekryd.sortpom</groupId>
        <artifactId>sortpom-maven-plugin</artifactId>
        <version>4.0.0</version>

        <configuration>
          <verifyFail>stop</verifyFail>
          <verifyFailOn>strict</verifyFailOn>
        </configuration>
      </plugin>

    </plugins>
  </build>
```

And the test didn't pass, but it should have...

**Comment by Ekryd** [comment]
Sorry for the silent pause. Work has been hectic recently. I'll check it out

**Comment by Ekryd** [comment]
Now, I see the problem! There is a very old parameter that is called `ignoreLineSeparators`.  Basically, if only the line breaks differ (including a missing line break at the end), the plugin will not sort the pom.
Try setting `ignoreLineSeparators` to false to make the SortPom plugin even pickier.

**Comment by Ekryd** [comment]
Please check out https://github.com/Ekryd/sortpom/pull/456
